### PR TITLE
Feature/issue 62/add logger

### PR DESCRIPTION
**Pull request by frey-perez**
Adding logger to config.py

### CODE REVIEW

File: src/wristpy/core/config.py [hunk 1]
```diff
@@ -9,3 +11,22 @@ class Settings(pydantic_settings.BaseSettings):
...
     VIGOROUS_THRESHOLD: float = 0.3
+
+    LOGGING_LEVEL: int = 20
+
+
+def get_logger() -> logging.Logger:
+    """Gets the wristpy logger."""
+    logger = logging.getLogger("wristpy")
+    if logger.handlers:
+        return logger
```
Comment (Asanto32) [inline src/wristpy/core/config.py:22]: This line is never covered. I know when we spoke in the meeting you had to some intricate stuff  with the LogCaptureFixture to get by the if statement, but should we include a test for when the handler already exists?

File: src/wristpy/core/config.py [hunk 2]
```diff
@@ -9,3 +11,22 @@ class Settings(pydantic_settings.BaseSettings):
     LIGHT_THRESHOLD: float = 0.03
     MODERATE_THRESHOLD: float = 0.1
     VIGOROUS_THRESHOLD: float = 0.3
+
+    LOGGING_LEVEL: int = 20
```
Comment (Asanto32) [inline src/wristpy/core/config.py:15]: Given @ReinderVosDeWael comment on slack, should this be set to 10 for DEBUG?

File: tests/unit/test_logger.py [hunk 3]
```diff
@@ -0,0 +1,19 @@
...
+def test_get_logger(caplog: pytest.LogCaptureFixture) -> None:
+    """Test the wristpy logger with level set to 20 (info)."""
+    logger = config.get_logger()
+
+    logger.debug("Debug message here.")
+    logger.info("Info message here.")
+    logger.warning("Warning message here.")
+
+    assert logger.getEffectiveLevel() == 20
+    assert "Debug message here" not in caplog.text
```
Comment (Asanto32) [inline tests/unit/test_logger.py:17]: If we change to DEBUG level, I believe this will have to change to

**Review by Asanto32 — COMMENTED** [review]
Some small comments here. My main question is whether we need to go populate the logger in the different modules (the primary candidates are most likely calibration, nonwear detection and sleep) in this PR? Or perhaps it will be mostly used in the runner/processor? We spoke about this in the meeting today but we weren't sure if that fits here or if it is a future task? Thoughts @ReinderVosDeWael

File: src/wristpy/core/config.py (hunk 2 above)
Comment (frey-perez) [reply src/wristpy/core/config.py:15]: In my conception of it, during the  normal course of running wristpy, the debug logs we write should be in the code, but not actually logging, so as not to flood the user with unnecessary information. When something goes wrong and we need to get in there and debug, you would switch the Settings().LOGGING_LEVEL to 10 yourself, and all those debug logs you wrote will now work. That was my reasoning for having 20 (INFO) as the default.

**Comment by ReinderVosDeWael** [comment]
Regarding @Asanto32 's comment, yes I'd include adding the basic logging statements in this PR. I'd keep it to debug/error statements in some likely offenders for things going wrong in the future, and we can add info statements to the runners.

File: src/wristpy/processing/analytics.py [hunk 4]
```diff
@@ -83,14 +85,17 @@ def run_sleep_detection(self) -> List[SleepWindow]:
...
+        logger.debug("Beginning sleep detection.")
         spt_window = self._spt_window(self.anglez)
         sib_periods = self._calculate_sib_periods(self.anglez)
         spt_window_periods = _find_periods(spt_window)
         sib_window_periods = _find_periods(sib_periods)
         sleep_onset_wakeup = self._find_onset_wakeup_times(
             spt_window_periods, sib_window_periods
         )
-
+        logger.debug(
```
Comment (ReinderVosDeWael) [inline src/wristpy/processing/analytics.py:96]: For all logging statements with variables in them, avoid f-strings (c.f. https://docs.astral.sh/ruff/rules/logging-f-string/ for an explanation of why). 

Personally, I prefer the *args method, but either *args or extra (both described in the above link) are fine. Optionally, you could enable Ruff's G-rules to be notified of this and several other issues.

File: src/wristpy/processing/calibration.py [hunk 5]
```diff
@@ -201,6 +217,7 @@ def _chunked_calibration(
             CalibrationError: If all possible chunks have been used and the calibration
                 process fails to get below the `min_calibration_error` threshold.
         """
+        logger.debug("running chunked calibration.")
```
Comment (ReinderVosDeWael) [inline src/wristpy/processing/calibration.py:220]: Minor thing; but you're mixing capitalized and non-capitalized logging statements. I'd suggest making a consistent choice here.

File: tests/unit/test_logger.py (hunk 3 above)
Comment (ReinderVosDeWael) [reply tests/unit/test_logger.py:17]: That's fine - if we do the test will fail and it can be modified at that point in time.

**Review by ReinderVosDeWael — CHANGES_REQUESTED** [review]
Looks good overall -- couple of minor comments. I'd also suggest adding logging to the file reader.

**Comment by ReinderVosDeWael** [comment]
What's the current status on this? Needs another review, still in the works?

**Comment by frey-perez** [comment]
> What's the current status on this? Needs another review, still in the works?
Ah forgot to ping, yes it's ready for review.

**Pull request merged**